localstack-core/localstack/services/stepfunctions/asl/component/common/path/output_path.py

```diff
@@ -20,3 +21,14 @@ def _eval_body(self, env: Environment) -> None:
             current_output = env.stack.pop()
             state_output = extract_json(self.output_path, current_output)
             env.inp = state_output
+
+
+class OutputPathContextObject(OutputPath):
+    def __init__(self, output_path: str):
+        output_path_tail = output_path[1:]
+        super().__init__(output_path=output_path_tail)
+
+    def _eval_body(self, env: Environment) -> None:
+        env.stack.pop()  # Discards the state output in favour of the context object path.
```

Comment:
To avoid a memory leak.
OutputPath is an EvalComponent. Its input, like for the others, is pushed onto the stack by the preceding expressions. If declared, OutputPath applies a reduction to its input (which is the current memory state) and sets it as input for the rest of the program. When choosing to source output from the context object, OutputPath does not apply the reduction to its input but to the context object. If we didn't consume the input, though, it would remain on the stack. That being said, we have a simple strategy in the Program component that clears the memory after each state. This means we could potentially avoid consuming the input here. However, when considering catch/retry or map and parallel states, memory usage would be worse than if we cleared it now, as the strategy in Program is intended as a failsafe mechanism.

Reply:
OK the above makes sense! Thanks for the detailed answer ✨ 

> That being said, we have a simple strategy in the Program component that clears the memory after each state.

Would be curious if there are any other strategies for this we could explore in future for this.